apps/schema/src/components/TableEditModal.vue

```diff
@@ -1,156 +1,196 @@
-        <InputString v-model="tableDraft.name" label="Table name" />
-        <InputText v-model="tableDraft.description" label="Table Description" />
-        <InputString
-          :list="true"
-          v-model="tableDraft.semantics"
-          label="semantics (comma separated list of IRI defining type, and/or keyword 'id')"
-        />
-      </LayoutForm>
-      <div v-else>ALTER IS NOT YET IMPLEMENTED</div>
-    </template>
-    <template v-slot:footer>
-      <MessageSuccess v-if="success">{{ success }}</MessageSuccess>
-      <MessageError v-if="graphqlError">{{ graphqlError }}</MessageError>
-      <ButtonAlt @click="$emit('close')">Close</ButtonAlt>
-      <ButtonAction v-if="tableDraft.oldName == false" @click="executeCommand"
-        >{{ action }}
-      </ButtonAction>
+      <ButtonAlt @click="cancel">Cancel</ButtonAlt>
+      <ButtonAction @click="close" :disabled="isDisabled">Apply</ButtonAction>
     </template>
   </LayoutModal>
 </template>
 
 <script>
-import {request} from 'graphql-request';
-
 import {
-  ButtonAction,
-  ButtonAlt,
   InputString,
   InputText,
-  LayoutForm,
   LayoutModal,
-  MessageError,
-  MessageSuccess,
-  SigninForm,
-  Spinner
-} from '@mswertz/emx2-styleguide';
+  IconAction,
+  ButtonAction,
+  MessageWarning,
+  InputSelect,
+  ButtonAlt,
+  deepClone,
+} from "molgenis-components";
 
 export default {
   components: {
-    MessageSuccess,
-    MessageError,
-    ButtonAction,
-    ButtonAlt,
     LayoutModal,
     InputString,
     InputText,
-    LayoutForm,
-    Spinner,
-    SigninForm
+    IconAction,
+    ButtonAction,
+    MessageWarning,
+    InputSelect,
+    ButtonAlt,
   },
   props: {
-    schema: String,
-    table: {type: Object, default: () => {}}
+    /** Existing Table metadata object entered as v-model. In case of a new table this should be left empty. */
+    value: {
+      type: Object,
+      required: false,
+    },
+    /** root table, used in case of subclasses */
+    rootTable: {
+      type: Object,
+      required: false,
+    },
+    /** schema, used for uniques check */
+    schema: {
+      type: Object,
+      required: true,
+    },
+    /** action, either 'add' or 'input */
+    operation: {
+      type: String,
+      default: "input",
+    },
+    /** type, either 'ontology' or nothing*/
+    tableType: {
+      type: String,
+    },
   },
   data: function () {
     return {
-      loading: false,
-      graphqlError: null,
-      success: null,
-      showLogin: false,
-      tableDraft: {}
+      /** copy of table metadata being edited now */
+      table: {},
+      /** whether modal is shown */
+      show: false,
     };
   },
   computed: {
     title() {
-      if (this.tableDraft.command === 'CREATE') {
-        return `Create table`;
-      } else {
-        return `Alter table '${this.table.name}'`;
+      return this.tableType === "ontology"
+        ? this.operation + " ontology definition"
```

Comment:
maybe use a string template here? it would be a bit more readible since you already use ? and : terniary operators.

```
 return this.tableType === "ontology"
        ? `${this.operation} ontology definition`
        : `${this.operation} table definition`;
```

Reply:
ok